homepage/homepage/content/docs/using-covalues/version-control.mdx

```diff
@@ -0,0 +1,240 @@
+  }
+
+  function handleTaskTitleChange(index: number, e: Event) {
+    const target = e.target as HTMLInputElement;
+    const task = project.current?.tasks[index];
+
+    // The task is also part of the branch because we used the `resolve` option
+    // with `tasks: { $each: true }`
+    // so the changes won't be visible on main until merged
+    task?.$jazz.set("title", target.value);
+  }
+</script>
+
+<form>
+  <!-- Edit form fields -->
+  <input 
+    type="text" 
+    value={project.current?.title || ''} 
+    oninput={handleTitleChange}
+  />
+  
+  {#each project.current?.tasks || [] as task, index}
+    <input 
+      type="text" 
+      value={task.title || ''} 
+      oninput={(e) => handleTaskTitleChange(index, e)}
+    />
+  {/each}
+</form>
+```
+</CodeGroup>
+</ContentByFramework>
+
+### Merging
+
+To merge a branch back to the main version, call `unstable_merge()`:
+
+<CodeGroup>
+```tsx 
+function handleSave() {
+  // Merge the branch changes back to main
+  project.$jazz.unstable_merge();
+  router.navigate("/");
+}
+```
+</CodeGroup>
+
+All the changes made to the branch will be merged into the main CoValue, preserving both author and timestamp.
+
+Calling `unstable_merge()` will merge every change done in the project and the loaded tasks.
+
+The merge is idempotent, so you can merge the same branch multiple times without changing the result.
+
+The merge operation is scoped, so if you call `unstable_merge()` on a task, only the changes to the task will be merged
+
+<CodeGroup>
+```tsx 
+function handleTaskSave(index: number) {
+  const task = project.tasks[index];
+  // Only the changes to the task will be merged
+  task?.$jazz.unstable_merge();
+}
+```
+</CodeGroup>
+
+
+## Conflict Resolution
+
+When conflicts occur (the same field is modified in both the branch and main), Jazz uses a "last writer wins" strategy:
+
+<CodeGroup>
+```tsx
+// Branch modifies priority to "high"
+branch.$jazz.applyDiff({ priority: "high" });
+
+// Meanwhile, main modifies priority to "urgent"
+originalProject.$jazz.applyDiff({ priority: "urgent" });
+
+// Merge the branch
+branch.$jazz.unstable_merge();
+
+// Main's value wins ("urgent") because it was written later
+console.log(originalProject.priority); // "urgent"
+```
+</CodeGroup>
+
+## Private branches
+
+When the owner is not specified, the branch has the same permissions as the main values.
+
+You can also create a private branch by providing a group owner.
+
+<CodeGroup>
+```tsx 
+// Create a private group for the branch
+const privateGroup = Group.create();
+
+const privateBranch = useCoState(Project, projectId, {
+  unstable_branch: { 
+    name: "private-edit", 
+    owner: privateGroup 
+  }
+});
+
+// Only members of privateGroup can see the branch content
+// The sync server cannot read the branch content
+```
+</CodeGroup>
+
+When a private branch is merged, the changes are merged into the main values, which keeps the privacy of the branch.
```

Comment:
I'm not sure I understand this. When the branch gets merged, all of the branch's history also gets merged onto `main`, right?

The above implies I could do something like
```ts
const privateBranch = useCoState(Project, projectId, {
  unstable_branch: { 
    name: "private-edit", 
    owner: privateGroup 
  }
});

privateBranch.$jazz.set('myPassword', 'hunter2');
privateBranch.$jazz.delete('myPassword')
privateBranch.$jazz.unstable_merge();
```

and my password would not be visible in the history of the `main` branch.

Maybe better just to remove this all together?